src/facade/dragonfly_connection.cc

```diff
@@ -580,6 +580,8 @@ void Connection::OnPostMigrateThread() {
     socket_->RegisterOnErrorCb([this](int32_t mask) { this->OnBreakCb(mask); });
   }
 
+  DCHECK(!dispatch_fb_.IsJoinable());
+
```

Comment:
Did you check what happens with the dispatch fiber when it get's a message during migration?

Reply:
It shouldn't? The connection migrates, before it does it joins the `dispatch_fb_`. At this point we got no messages and we are the active fiber on the thread we are executing. Then we suspend on the current thread and we continue our execution on the thread we migrate. During that time the `dispatch_fb_` should stay `joined` (that is, it couldn't be restarted) since no other operation can or should really wake it -- thinking of ACL's here(since the connection won't even show up when it gets removed by the thread local connection table and when it becomes available again on the thread local of the migrated thread it will be the active fiber (so no other fiber can wake up its queue). 


Am I missing something?

Reply:
Theoretically... we check that no subscriptions are active when we migrate and no command is ongoin, so there shouldn't be any, right.

We have this dcheck to catch it if it does. Maybe there will be a reason do so in the future and we'll forget this is a potential bug 🙂

Reply:
@kostasrim  @dranikpg  it is a bug. 
See https://github.com/dragonflydb/dragonfly/actions/runs/9332324376
`F20240601 18:46:34.649338 35021 dragonfly_connection.cc:570] Check failed: !dispatch_fb_.IsJoinable() `

Reply:
can `SendAsync` be called from another connection? In that case it would kick of the fiber. Correct @dranikpg ?

Reply:
> can SendAsync be called from another connection? 

Sure it can, pubsub as an example. I had a feeling that's something's wrong here, that's why I asked.

We migrate only when we don't have any subscriptions. The question is whether ACL messages or checkpoint messages see us on the new or old thread during migration

Reply:
We `should` check this -- I can -- but once I wrap up the current staff. 

> can SendAsync be called from another connection? 

Both connection fiber and dispatch fiber run on the same thread so during a migration the connection fiber is `active` so `SendAsync` should `happen after` the migration and therefore it should have the correct semantics? 

Maybe I am missing something... Either way we have a track record of having issues around this code and we should take care of it. 

I will ping here once I get a chance

Reply:
It could be that the Fiber is migrated but was not switched to yet. Meanwhile another producer-fiber find the connection on a new thread and calls SendAsync. The migrated fiber finaly resumes and finds out its dispatch_fb_ has started. Seems like a valid scenario.